packages/kbn-field-types/src/kbn_field_types_factory.ts

```diff
@@ -21,6 +21,7 @@ export const createKbnFieldTypes = (): KbnFieldType[] => [
     esTypes: [
       ES_FIELD_TYPES.STRING,
       ES_FIELD_TYPES.TEXT,
+      ES_FIELD_TYPES.MATCH_ONLY_TEXT,
```

Comment:
@elastic/kibana-app-services -- in leveraging the `castEsToKbnFieldTypeName` utility I found this missing [ES type](https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html#match-only-text-field-type). I imagine this is fine as it's a sibling type to `text`, but was curious here since `sortable: true,` is set above even though it includes `ES_FIELD_TYPES.TEXT` (which is not sortable) as well?

Reply:
Also, I had [previously updated](https://github.com/elastic/kibana/pull/131475/files#diff-d79a8297783f3177da25dd13fe807425d9136a0e235fe170f7c0a61f2448dacaR23) `FieldIcon` to support `match_only_text` as well, but this was before I noticed the distinction between `ES_FIELD_TYPES` and `KBN_FIELD_TYPES` with the `FieldIcon` component. Seems the `FieldIcon` component kind of supports both (with at least `keyword` coming from ES_FIELD_TYPES) -- should I undo my previous change in support of `match_only_text` there as well, or it fine to leave it as-is?

Reply:
Moved to dedicated PR -- please see https://github.com/elastic/kibana/pull/133690